R/methods-get_filter_call.R

```diff
@@ -1,24 +1,40 @@
+setOldClass(c("teal_slice", "list"))
+
 #' Get filter call
 #'
 #' Get filter call
+#'
 #' @inheritParams filter_panel_methods
 #' @inheritSection filter_panel_methods Supported data types
-#' @export
-get_filter_call <- function(data, states_list) {
-  if (length(states_list) > 0L) {
-    UseMethod("get_filter_call")
-  }
-}
-
+#'
+#' @name get_filter_call
 #' @rdname get_filter_call
+#' @aliases get_filter_call-ANY-method
+#' @aliases get_filter_call-data.frame-method
+#' @aliases get_filter_call-DataFrame-method
+#' @aliases get_filter_call-matrix-method
+#' @aliases get_filter_call-Matrix-method
+#' @aliases get_filter_call-SummarizedExperiment-method
+#' @aliases get_filter_call-MultiAssayExperiment-method
+#'
 #' @export
-get_filter_call.default <- function(data, states_list) {
-  stop("get_filter_call not implemented for class ", class(data), call. = FALSE)
-}
+#'
+# get_filter_call generic ----
+setGeneric("get_filter_call", function(data, states_list) {
+  NULL
+})
+
+## default method ----
+setMethod("get_filter_call", c(data = "ANY"), function(data, states_list) {
+  if (length(states_list) == 0L) NULL
```

Comment:
```suggestion
  if (length(states_list) == 0L) return(NULL)
```

Reply:
By `NEST` policy we avoid explicit `return` calls at the end of functions.

Reply:
Take a look at the code as these are not the last expression on the function and will do nothing.

That is, it won't return `NULL` and just continue to the next expression

Reply:
This is valid to every other comment.

If you prefer, wrap the following expressions around an `else` and keep it without `return()` call

Reply:
You're right.
Now that I look at it, I'm not sure why it is written this way, actually.